operator/controllers/keptnworkloadinstance/reconcile_deploymentstate.go

```diff
@@ -42,24 +40,32 @@ func (r *KeptnWorkloadInstanceReconciler) reconcileDeployment(ctx context.Contex
 	return workloadInstance.Status.DeploymentStatus, nil
 }
 
-func (r *KeptnWorkloadInstanceReconciler) isReplicaSetRunning(ctx context.Context, resource klcv1alpha1.ResourceReference, namespace string) (bool, error) {
-	replica := &appsv1.ReplicaSetList{}
-	if err := r.Client.List(ctx, replica, client.InNamespace(namespace)); err != nil {
-		return false, err
-	}
-	for _, re := range replica.Items {
-		if re.UID == resource.UID {
-			replicas, err := r.getDesiredReplicas(ctx, re.OwnerReferences[0], namespace)
-			if err != nil {
-				return false, err
-			}
-			if re.Status.ReadyReplicas == replicas {
-				return true, nil
-			}
-			return false, nil
+func (r *KeptnWorkloadInstanceReconciler) isOwnerRunning(ctx context.Context, resource klcv1alpha1.ResourceReference, namespace string) (bool, error) {
+
+	var replicas *int32
+	var desiredReplicas int32
+	switch resource.Kind {
+	case "ReplicaSet":
+		dep := appsv1.ReplicaSet{}
+		err := r.Client.Get(ctx, types.NamespacedName{Name: resource.Name, Namespace: namespace}, &dep)
+		if err != nil {
+			return false, err
+		}
+		replicas = dep.Spec.Replicas
```

Comment:
Can we please check here that the UID is the same as in the resource? Just to be absolutely sure, because searching by name isn't the best strategy

Reply:
k8s names are unique, and the get will always return the owner of the deployment